addons/website_livechat/models/im_livechat_channel.py

```diff
@@ -43,12 +43,33 @@ def _get_livechat_discuss_channel_vals(self, /, *, chatbot_script=None, agent=No
     def create(self, vals_list):
         channels = super().create(vals_list)
         if self.env.context.get("create_from_website"):
-            for channel in channels:
+            website = self.env['website'].get_current_website()
+            bot_record = self.env.ref("im_livechat.chatbot_script_welcome_bot", raise_if_not_found=False)
+            bot = bot_record.id if bot_record else False
```

Comment:
Always prefer to work with recordset rather than id whenever possible.
`bot_record` is `bot`, and can fallback to empty bot recordset if not found.

Using `bot.id` below is fine even when empty (it will return False)